pkg/sql/logictest/testdata/logic_test/pg_catalog

```diff
@@ -2665,3 +2665,10 @@ WHERE
   AND attname = 'x'
 ----
 'howdy'::STRING
+
+# Regression test for limits on virtual index scans. (#53522)
+
+query T
+SELECT attname FROM pg_attribute WHERE attrelid = 't47285'::REGCLASS::OID LIMIT 1
```

Comment:
oddly enough, it looks like this query is _not_ affected by the original issue. it seems like the casting makes the query avoid hitting the problem. is there a way to write a test without casting? idk how, so if not, the tests lgtm

example from me testing locally with v20.2.0-alpha:
```
root@:26257/defaultdb> SELECT attrelid, attname FROM pg_attribute WHERE attrelid = 't'::REGCLASS::OID limit 1;
  attrelid | attname
-----------+----------
       215 | name
(1 row)

Time: 66.961ms

root@:26257/defaultdb> SELECT attrelid, attname FROM pg_attribute WHERE attrelid = 215 limit 1;
  attrelid |    attname
-----------+-----------------
       215 | name
       215 | username
       215 | security_level
       215 | last_login
       215 | locked
       215 | rowid
(6 rows)
```